solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -191,6 +196,70 @@ pub mod solana_ibc {
         chain.set_stake((validator).into(), amount)
     }
 
+    /// Sets up new fee collector proposal which wont be changed until the new fee collector
+    /// calls `accept_fee_collector_change`. If the method is called for the first time, the fee
+    /// collector would just be set without needing for any approval.
+    ///
+    pub fn setup_fee_collector<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, SetupFeeCollector<'info>>,
+        new_fee_collector: Pubkey,
+    ) -> Result<()> {
+        let private_storage = &mut ctx.accounts.storage;
+
+        let signer = ctx.accounts.fee_collector.key();
+
+        if private_storage.fee_collector == Pubkey::default() {
+            private_storage.fee_collector = new_fee_collector;
+        } else if signer == private_storage.fee_collector {
+            private_storage.new_fee_collector_proposal = Some(new_fee_collector)
+        } else {
+            return Err(error!(error::Error::InvalidFeeCollector));
+        }
+
+        Ok(())
+    }
+
+    pub fn accept_fee_collector_change<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, SetupFeeCollector<'info>>,
+    ) -> Result<()> {
+        let private_storage = &mut ctx.accounts.storage;
+
+        let signer = ctx.accounts.fee_collector.key();
+
+        if let Some(new_admin) = private_storage.new_fee_collector_proposal {
+            if signer != new_admin {
+                return Err(error::Error::InvalidFeeCollector)
+                    .map_err(|err| error!((&err)));
+            }
+            private_storage.fee_collector = new_admin;
+            private_storage.new_fee_collector_proposal = None;
+        } else {
+            return Err(error!(error::Error::FeeCollectorChangeProposalNotSet));
+        }
+
+        Ok(())
+    }
+
+    pub fn collect_fees<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, CollectFees<'info>>,
+    ) -> Result<()> {
+        let fee_collector_balance = ctx.accounts.fee_account.lamports();
```

Comment:
We need to retain enough lamports for the account to stay rent exempt, no?

And if we want to support refunds, we probably also want to keep some more balance on the account.

Reply:
thats why we are keeping 1 SOL which seems good enough.

Reply:
Ah, ok, I now see what `MINIMUM_FEE_TO_COLLECT` means. I think the constant has a confusing name.  It should be `MINIMUM_FEE_ACCOUNT_BALANCE` or some such.  Might still consider

```rust
    let fee_account = &ctx.accounts.fee_account;
    let minimum_balance =
        Rent::get()?.minimum_balance(fee_account.data_len())? +
        MINIMUM_FEE_ACCOUNT_BALANCE;
    let available_balance = fee_account.try_borrow_mut_lamports()?;
    if **available_balance > minimum_balance {
        **ctx.accounts.fee_collector.try_borrow_mut_lamports()? +=
            (**available_balance - minimum_balance);
        **available_balance = minimum_balance;
        Ok(())
    } else {
        Err(error!(error::Error::InsufficientFeesToCollect))
    }
```

though now accounting for rent is less important.

Reply:
but i think it makes sense to have the above.